recipes/humann/meta.yaml

```diff
@@ -0,0 +1,90 @@
+{% set name = "humann" %}
+{% set version = "3.0.0a4" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 300e91dee00c5f4cb0f5a529164be1d7332fdbd288685a865074f48c0cb3459f
+
+build:
+  number: 0
+  noarch: python
+  entry_points:
+    - humann = humann.humann:main
+    - humann3 = humann.humann:main
+    - humann_databases = humann.tools.humann_databases:main
+    - humann3_databases = humann.tools.humann_databases:main
+    - humann_config = humann.tools.humann_config:main
+    - humann_join_tables = humann.tools.join_tables:main
+    - humann_split_table = humann.tools.split_table:main
+    - humann_rename_table = humann.tools.rename_table:main
+    - humann_renorm_table = humann.tools.renorm_table:main
+    - humann_regroup_table = humann.tools.regroup_table:main
+    - humann_infer_taxonomy = humann.tools.infer_taxonomy:main
+    - humann_humann1_kegg = humann.tools.humann_humann1_kegg:main
+    - humann_rna_dna_norm = humann.tools.rna_dna_norm:main
+    - humann_strain_profiler = humann.tools.strain_profiler:main
+    - humann_reduce_table = humann.tools.reduce_table:main
+    - humann_unpack_pathways = humann.tools.merge_abundance:main
+    - humann_test = humann.tests.humann_test:main
+    - humann_build_custom_database = humann.tools.build_custom_database:main
+    - humann_genefamilies_genus_level = humann.tools.genefamilies_genus_level:main
+    - humann_split_stratified_table = humann.tools.split_stratified_table:main
+    - humann_associate = humann.tools.humann_associate:main
+    - humann_barplot = humann.tools.humann_barplot:main
+    - humann_benchmark = humann.tools.humann_benchmark:main
+  script: "{{ PYTHON }} -m pip install . -vv"
```

Comment:
Seems that `setup.py` downloads and compiles quite a bit of software:

- glpk-4.55.tar.gz
- minpath1.2
- boost_1_57
- diamond 0.9.24
- bowtie2-2.2.3

If I get it right the option `--bypass-dependencies-install` should avoid this. These should be added as requirements. 

Apart from minpath (https://omics.informatics.indiana.edu/MinPath/, https://github.com/mgtools/MinPath) everything seems available for conda. 

Diamond should probably restricted to a version <1.

Reply:
I updated to use `bypass-dependencies-install` and avoid installation of `glpk`, `diamond` and `bowtie`. 
For `minpath` is still installed there: it could be replaced by using `replace-dependencies-install` but I am afraid it will break the tool afterwards as `minpath` and the specific version used are hardcoded several times in `humann` code.

Reply:
I think the best would be to add minpath1.2 to bioconda .. and then to patch setup.py to not install it ..

Reply:
I am not 100% sure what to do about that

HUMAnN installs by default MinPath1.2, which seems to be Python 2.7. Should we add a Python 2.7 package?
I am not sure how HUMAnN can use that without conflict between Python versions.

Since then, MinPath has been migrated to GitHub (https://github.com/mgtools/MinPath) and seems to support Python 3.
But:
1. no release has been created there
2. not sure if HUMAnN will support a newest version.

What would be the best approach there?

I see that in the conda package (https://github.com/biobakery/conda-biobakery/tree/master/recipes/humann) they provided through their conda channel that they leave the installation on MinPath to humann

Reply:
> MinPath1.2, which seems to be Python 2.7. Should we add a Python 2.7 package?

I think we can keep it as is for now. the humann github lists python >3.7 as requirement. https://github.com/biobakery/humann#software

Maybe add minpath 1.2 to the requirements, but comment it and mark it as TODO. 

> I see that in the conda package (https://github.com/biobakery/conda-biobakery/tree/master/recipes/humann) they provided through their conda channel that they leave the installation on MinPath to humann

Should be fine for now. At least minpath is python and not binary - so we can keep it `noarch: python`.

Btw. https://github.com/biobakery/humann#software lists quite some more requirements .. but we can leave it as is for now.